docs/docs/index.md

```diff
@@ -13,19 +13,20 @@ The framework supports the following types of agents:
 
 Key features of Koog include:
 
-- **Pure Kotlin implementation**: Build AI agents entirely in natural and idiomatic Kotlin.
-- **MCP integration**: Connect to Model Control Protocol for enhanced model management.
-- **Embedding capabilities**: Use vector embeddings for semantic search and knowledge retrieval.
-- **Custom tool creation**: Extend your agents with tools that access external systems and APIs.
-- **Ready-to-use components**: Speed up development with pre-built solutions for common AI engineering challenges.
-- **Intelligent history compression**: Optimize token usage while maintaining conversation context using various pre-built strategies.
+- **Multiplatform development**: Deploy agents across JVM, JS, WasmJS, Android, and iOS targets using Kotlin Multiplatform.
+- **Reliability and fault-tolerance**: Handle failures with built-in retries and restore full agent state machines with the agent persistence feature.
+- **Intelligent history compression**: Optimize token usage while maintaining context in long-running conversations using advanced built-in history compression techniques.
+- **Enterprise-ready integrations**: Utilize integration with popular JVM frameworks such as Spring Boot and Ktor to embed Koog into your applications.
+- **Observability with OpenTelemetry exporters**: Monitor and debug applications with built-in support for popular observability providers (W&B Weave, Langfuse).
+- **LLM switching and seamless history adaptation**: Switch to a different LLM at any point without losing the existing conversation history, or reroute between multiple LLM providers.
+- **Integration with JVM and Kotlin applications**: Build AI agents with an idiomatic, type-safe Kotlin DSL designed specifically for JVM and Kotlin developers.
+- **Model Context Protocol integration**: Use Model Context Protocol (MCP) tools in AI agents.
+- **Knowledge retrieval and memory**: Retain and retrieve knowledge across conversations using vector embeddings, semantic search, ranked document storage, and shared agent memory.
```

Comment:
Do we really have such deep integrations? AFAIK there's only `rag` set of modules with more low-level API to get embeddings and calculate distances between vectors. Or am I missing something here?

Reply:
My mistake, seems like we do have some more advanced integrations